pkg/querier/timeline/timeline.go

```diff
@@ -18,56 +18,57 @@ func New(series *v1.Series, startMs int64, endMs int64, durationDeltaSec int64)
 	points := series.GetPoints()
 	res := make([]uint64, len(points))
```

Comment:
nit: I wonder if we can pre allocate the whole buffer here with zero length and `sizeToBackfill(startMs, endMs, durationDeltaSec)` capacity

And then down the line instead of alloc + append backfill buffer, just extend the length of the res. 

This is probably premature optimization. feel free to ignore the comment.